libraries/macos_user.rb

```diff
@@ -1,9 +1,13 @@
 module MacOS
   module MacOSUserHelpers
     def kcpassword_hash(password)
-      bits = magic_bits
+      bits = if password.nil?
+               nil_magic_bits
+             else
+               magic_bits
+             end
       obfuscated = []
-      padded(password).each do |char|
+      padded(password.to_s).each do |char|
```

Comment:
As far as I know `.to_s` on `nil` in Ruby returns `""`. This effectively does nothing if `password.nil?` is true.

Reply:
Ruby raises an exception that there is no *implicit* conversion of nil to String without this.